docs/source/openapi.json

```diff
@@ -1121,23 +1121,21 @@
             "type": "number"
           },
           "min": {
-            "type": "number"
+            "type": { "anyOf": [{ "type": "number" }, { "type": "null" }] }
```

Comment:
Can the stats by partially null (eg "min", "max"), and partially non-null (eg "mean" and "median")? Or is it always: all null, or all non-null?

Reply:
it's either all null (if all values in column are null) or all non-null

Reply:
OK. So... I think it would be easier to handle on the client's side if we change the type a bit: return `null` or an object with all the stats as "number" (ie: no individually optional number). Do you think it would make sense?

Reply:
you mean return `null` as a whole response instead of the object with only two filled fields (`nan_count`, `nan_proportion`)?

Reply:
yes. Because otherwise, the client has to assume that we can have min=null and max=10, which is not true.

Reply:
right... yes, that sounds very reasonable. but i'm not sure it is aligned with the server logic.
we have:
```
class NumericalStatisticsItem(TypedDict):
    nan_count: int
    nan_proportion: float
    min: Optional[float]  # might be None in very rare cases when the whole column is only None values
    max: Optional[float]
    mean: Optional[float]
    median: Optional[float]
    std: Optional[float]
    histogram: Optional[Histogram]


class CategoricalStatisticsItem(TypedDict):
    nan_count: int
    nan_proportion: float
    no_label_count: int
    no_label_proportion: float
    n_unique: int
    frequencies: dict[str, int]


class BoolStatisticsItem(TypedDict):
    nan_count: int
    nan_proportion: float
    frequencies: dict[str, int]


SupportedStatistics = Union[NumericalStatisticsItem, CategoricalStatisticsItem, BoolStatisticsItem]

class StatisticsPerColumnItem(TypedDict):
    column_name: str
    column_type: ColumnType
    column_statistics: SupportedStatistics
```
In response `column_statistics` is always non-null, does it make sense to make it nullable? it would mean then that it can be null only in a specific case - for numerical columns with all-None values. `CategoricalStatisticsItem` and `BoolStatisticsItem` don't do the same because they have all other values set to some non-null values.

btw i also realized now that front doesn't show anything for such numerical cases, while i think it should as it does for categories and bool:
https://huggingface.co/datasets/polinaeterna/only_null

Reply:
-> https://github.com/huggingface/moon-landing/issues/9216 (internal)

Reply:
hmmm, I think I was wrong. It's OK to have all the values as `Optional[float]` / `Optional[Histogram]` after all